smack-core/src/main/java/org/jivesoftware/smack/packet/Stanza.java

```diff
@@ -502,6 +650,19 @@ protected void addCommonAttributes(XmlStringBuilder xml) {
     }
 
     /**
+     * Add custom attributes
+     *
+     * @param xml
+     */
+    protected void addCustomAttributes(XmlStringBuilder xml) {
+        if(customAttributes != null) {
```

Comment:
I would go for `if (customAttributes == null) return;` here, to avoid a level of indentation.